thirdeye-ui/src/app/pages/alerts-anomalies-page/alerts-anomalies-page.component.tsx

```diff
@@ -110,7 +138,7 @@ export const AlertsAnomaliesPage: FunctionComponent = () => {
     useEffect(() => {
         // Fetched alert changed, fetch alert evaluation
         fetchData();
-    }, [alert, startTime, endTime, enumerationItemIdStr]);
+    }, [alert, startTime, endTime, enumerationItemIdStr, showAllAnomalies]);
```

Comment:
We should prevent having to make another network call for the same data if we're just applying a local filter. I suggest refactoring so that another state variable holds the filtered array of anomalies and updating that when `showAllAnomalies` boolean changes

Reply:
Changed the component to compute `uiAnomalies` by filtering all `anomalies`. However, I had to add a `fetchData()` to `handleAnomalyDelete` since it was modifying `uiAnomalies` locally, and preserving that behavior would complicate the local state (will have to keep track of all anomalies deleted). 

Should we retain the original `handleAnomalyDelete` behavior, or is it fine to make the fetch calls? (anomaly deletion might not be such a common operation, and they are also batched.)